File: ConstructionMachinery/WebAPI/Models/UserUpdateModel.cs
using System.ComponentModel.DataAnnotations;

namespace WebAPI.Models
{
    public class UserUpdateModel
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Email { get; set; }
        public string Phone { get; set; }

        [DataType(DataType.Password)]
        public string Password { get; set; }
    }
}

File: WebApi/ClientApp/src/app/models/UserUpdateModel.ts
export class UserUpdateModel {
    id: number;
    name: string;
    email: string;
    phone: string;
    password: string

    public constructor(_id: number, _name: string, _email: string, _phone: string, _password: string) {
        this.id = _id,
        this.name = _name,
        this.email = _email,
        this.phone = _phone;
        this.password = _password;
    }
}